libpod/container_internal.go

```diff
@@ -1152,9 +1152,27 @@ func (c *Container) restartWithTimeout(ctx context.Context, timeout uint) (err e
 	c.newContainerEvent(events.Restart)
 
 	if c.state.State == define.ContainerStateRunning {
+		conmonPID := c.state.ConmonPID
 		if err := c.stop(timeout); err != nil {
 			return err
 		}
+		// Old versions of conmon have a bug where they create the exit file before
+		// closing open file descriptors causing a race condition when restarting
+		// containers with open ports since we cannot bind the ports as they're not
+		// yet closed by conmon.
+		//
+		// Killing the old conmon PID is ~okay since it forces the FDs of old conmons
+		// to be closed, while it's a NOP for newer versions which should have
+		// exited already.
+		if conmonPID != 0 {
+			p, err := os.FindProcess(conmonPID)
+			if err != nil {
+				logrus.Debugf("error finding conmon process: %v", err)
+			}
+			if err := p.Kill(); err != nil {
```

Comment:
Might this segfault if the first `err` was non-nil? I don't know if a process is a pointer, but it seems like it might be.

Reply:
Definitely an issue. Thanks for having another look.